test/metatx/MinimalForwarder.test.js

```diff
@@ -22,150 +31,331 @@ contract('MinimalForwarder', function (accounts) {
         { name: 'value', type: 'uint256' },
         { name: 'gas', type: 'uint256' },
         { name: 'nonce', type: 'uint256' },
+        { name: 'deadline', type: 'uint48' },
         { name: 'data', type: 'bytes' },
       ],
     };
-  });
 
-  context('with message', function () {
-    const tamperedValues = {
-      from: accounts[0],
-      to: accounts[0],
-      value: web3.utils.toWei('1'),
-      nonce: 1234,
-      data: '0x1742',
+    this.alice = Wallet.generate();
+    this.alice.address = web3.utils.toChecksumAddress(this.alice.getAddressString());
+
+    this.blockNumber = await time.latestBlock();
+    this.request = {
+      from: this.alice.address,
+      to: constants.ZERO_ADDRESS,
+      value: '0',
+      gas: '100000',
+      nonce: Number(await this.forwarder.nonces(this.alice.address)),
```

Comment:
In [ERC2771Context.test.js](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4346/files?diff=split&w=0#diff-296c2f0ca2b388a964973951bfce6a90eefa2b24df65564389dd7f944b6d896e) you do `toString()` and here number. IMO we should do the same everywhere.